### PR TITLE
Add user interests and helper enums AgeRange and GenderPreference

### LINKED ISSUE #40: Create entity and supporting classes for UserInterest

### PR DESCRIPTION
**Pull request by amoscalamida**
closes #40

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private int ageRangeMax;
+
+    @Enumerated(EnumType.STRING)
+    private GenderPreference genderPreference;
+
+    @ElementCollection(targetClass = ActivityCategory.class)
+    @CollectionTable(name = "USER_INTERESTS_ACITIVITY", joinColumns = @JoinColumn(name = "userInterestsId"))
+    @Column
+    @Enumerated(EnumType.STRING)
+    private Set<ActivityCategory> activityInterests;
```
Comment (AleksandarRistic99) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java:38]: isch das, damit mehreri AcvitityCategories chene inere collection gspeichert werde?

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private int ageRangeMin;
+
+    @Column
+    private int ageRangeMax;
+
+    @Enumerated(EnumType.STRING)
+    private GenderPreference genderPreference;
+
+    @ElementCollection(targetClass = ActivityCategory.class)
+    @CollectionTable(name = "USER_INTERESTS_ACITIVITY", joinColumns = @JoinColumn(name = "userInterestsId"))
```
Comment (AleksandarRistic99) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java]: und wieso bruchts do en join?

**Comment by AleksandarRistic99** [comment]
Sösch findis guet 👍

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java (hunk 2 above)
Comment (amoscalamida) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java]: Will d Collection vo Interests inere separate table muess gspeicheret werde

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java (hunk 1 above)
Comment (amoscalamida) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java:38]: Jep

Comment (AleksandarRistic99) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java:38]: okee merci,  @SamuelBruegger vlt chentsch s problem met de List mit de Strings und de List mit de LocalDateTime be de SchedulingSession eso löse wie das de Amos do gmacht het

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java (hunk 2 above)
Comment (AleksandarRistic99) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java]: easii danke

**Review by AleksandarRistic99 — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java (hunk 1 above)
Comment (Taremeh) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java:38]: Yes, using it this way will probably fix the issue. I've created a task #63 and will add it to my pull request.
Btw I have also created a wiki page regarding Hibernate JPA, so we can use it as a quick reference when doing Annotation work in the backend: https://github.com/sopra-fs21-group-16/mth-server/wiki/JPA-Hibernate-Docs

**Review by Taremeh — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java (hunk 2 above)
Comment (Taremeh) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java]:
```suggestion
    @CollectionTable(name = "USER_INTERESTS_ACTIVITY", joinColumns = @JoinColumn(name = "userInterestsId"))
```

Comment (amoscalamida) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/entities/UserInterests.java]: Thx 😉

**Pull request merged**